### PR TITLE
Parameter to delete previous migration

### PR DESCRIPTION
**Pull request by Kriechi**
This new parameter will drop all migration-tables and delete the migration-directory if it already exists.
I'm not quite sure if the location for calling is ok, maybe a separate step should be implemented - or not.

Issue 13 - I think thats what asked for.

### CODE REVIEW

File: trunk/admin/controllers/ajax.php [hunk 1]
```diff
@@ -40,6 +44,37 @@ function getParams()
...
+	 * Deletes all tables with new-prefix and the migration folder
+	 *
+	 * @return	none
+	 * @since	1.X
+	 */
+	function deletePreviousMigration()
+	{
+		$jupgrade = new jUpgrade;
+
+		// delete all tables of previous migration
```
Comment (fastslack) [inline trunk/admin/controllers/ajax.php]: We are doing this into the cleanup() method

File: trunk/jupgrade.xml [hunk 2]
```diff
@@ -1,122 +1,119 @@
...
+    <param type="spacer" default="&lt;b&gt;Debug&lt;/b&gt;"/>
+    <param name="debug_php" type="list" default="0" label="Enable migration debug" description="">
+      <option value="0">No</option>
+      <option value="1">Yes</option>
+    </param>
+    <param name="debug_js" type="list" default="0" label="Enable Mootools version debug" description="">
+      <option value="0">No</option>
+      <option value="1">Yes</option>
+    </param>
+    <param type="spacer" default="&lt;b&gt;Debug&lt;/b&gt;"/>
```
Comment (fastslack) [inline trunk/jupgrade.xml]: We need to have other title than "Debug"

**Comment by Kriechi** [comment]
spacer-text was a copy-paste mistake.
moved the function-call to the end of the cleanup-function.
also added error-checking for query-errors.

But I'm not quite sure if we could then remove ajax.php:238-254 completely?

**Comment by Kriechi** [comment]
Well, that last one was rubish,...
Doing it in the cleanup-section isn't going to work, because at this point the new installation is already extracted and in place.

**Comment by fastslack** [comment]
The only thing you must to do is to remove the DROP tables because we are doing that on cleanup() method... please make a new pull request, this is very buggy.

**Comment by mahagr** [comment]
I don't like the pull request because of it changes even those lines which haven't been affected -- thus making it impossible to see the changes.

Please configure your editor to keep line endings as they were before.

**Pull request merged**